sql/core/src/main/scala/org/apache/spark/sql/execution/joins/CartesianProduct.scala

```diff
@@ -34,7 +34,15 @@ case class CartesianProduct(left: SparkPlan, right: SparkPlan) extends BinaryNod
     val leftResults = left.execute().map(_.copy())
     val rightResults = right.execute().map(_.copy())
 
-    leftResults.cartesian(rightResults).mapPartitions { iter =>
+    val cartesianRdd = if (leftResults.partitions.size > rightResults.partitions.size) {
+      rightResults.cartesian(leftResults).mapPartitions { iter =>
+        iter.map(tuple => (tuple._2, tuple._1))
+      }
+    } else {
+      leftResults.cartesian(rightResults)
+    }
+
+    cartesianRdd.mapPartitions { iter =>
       val joinedRow = new JoinedRow
```

Comment:
@hvanhovell Yes, use sizeInBytes is better, but also have a problem, if leftResults only have 1 record and this record size are big, and rightResults have many records and these records total size are small, then at this scenario will cause worse performance. The best way is we check the total records for the partition, but now we can not get it.